registry/storage/driver/swift/swift_test.go

```diff
@@ -149,4 +148,24 @@ func TestEmptyRootList(t *testing.T) {
 			t.Fatalf("unexpected string in path: %q != %q", path, storagedriver.PathRegexp)
 		}
 	}
+
+	// Create an object with a path nested under the existing object
+	err = rootedDriver.PutContent(ctx, filename+"/file1", contents)
```

Comment:
@aaronlehmann The storagedriver.Delete method should behave the exact same way, whether or not a specific backend feature is enabled or not.

> Note that this
coverage was added to the swift unit test rather than the generic driver
testsuite because the test case is not applicable to all storage
drivers. For example, with the filesystem driver it's not possible to
have a file underneath another file.

No, this is not possible, but it is for directories and other s3 configurations. If I `Delete("a")`, any child of "a" should be removed.

We should consider adding a test, even if it is in a separate PR. Let's get this merged and have a PR to move this into the main testsuite.